src/Pay/Adapter/Stripe.php

```diff
@@ -0,0 +1,180 @@
+     */
+    public function getName(): string {
+        return 'Stripe';
+    }
+  
+    /**
+     * Make a purchase request
+     */
+    public function purchase(int $amount, string $customerId, string $cardId = null, array $additonalParams = []): array {
+        $requestBody = [
+            'customer' => $customerId,
+            'amount' => $amount,
+            'currency' => $this->currency,
+        ];
+        $requestBody = array_merge($requestBody, $additonalParams);
+        if(!empty($cardId)) {
+            $requestBody['source'] = $cardId;
+        }
+        $res = $this->execute('POST', '/charges', $requestBody);
+        return $res;
+    }
+  
+    /**
+     * Refund payment
+     */
+    public function refund(string $paymentId, int $amount = null) : array {
+        $path = '/refunds';
+        $requestBody = ['charge' => $paymentId];
+        if($amount != null) {
+            $requestBody['amount'] = $amount;
+        }
+        return $this->execute('POST', $path, $requestBody);
+    }
+
+    /**
+     * Add a credit card for customer
+     */
+    public function createCard(string $customerId, string $cardId): array {
+        $path = '/customers/' . $customerId . '/sources';
+        return $this->execute('POST', $path, ['source' => $cardId]);
+    }
+
+    /**
+     * List cards
+     */
+    public function listCards(string $customerId): array {
+        $path = '/customers/' . $customerId . '/sources';
+        return $this->execute('GET', $path);
+    }
+
+    /**
+     * Update card
+     */
+    public function updateCard(string $customerId, string $cardId, string $name = null, int $expMonth = null, int $expYear = null, array  $billingDetails = null ): array {
+        $path = '/customers/' . $customerId . '/sources/' . $cardId;
+        $requestBody = [];
+        if(!empty($name)) {
+            $requestBody['name'] = $name;
+        }
+        if(!empty($expMonth)) {
+            $requestBody['exp_month'] = $expMonth;
+        }
+        if(!empty($expYear)) {
+            $requestBody['exp_year'] = $expYear;
+        }
+        return $this->execute('POST', $path, $requestBody);
+    }
+
+    /**
+     * Get a card
+     */
+    public function getCard(string $customerId, string $cardId): array {
+        $path = '/customers/' . $customerId . '/sources/' . $cardId;
+        return $this->execute('GET', $path);
+    }
+
+    /**
+     * Delete a credit card record
+     */
+    public function deleteCard(string $customerId, string $cardId) : bool {
+        $path = '/customers/' . $customerId . '/sources/' . $cardId;
+        $res =  $this->execute('DELETE', $path);
+        return $res['deleted'] ?? false;
+    }
+  
+    /**
+     * Add new customer in the gateway database
+     * returns the id of the newly created customer
+     * 
+     * @throws Exception
+     */
+    public function createCustomer(string $name, string $email, array $billingDetails = [], string $paymentMethod = null) : array {
```

Comment:
This method is supposed to return an ID, so I believe we should change the return type to better reflect that.

Reply:
The description error, I am returning the newly created customer record not only the ID